thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -71,16 +74,18 @@ public int compare(RawAnomalyResultDTO o1, RawAnomalyResultDTO o2) {
         populateMergedResult(mergedAnomaly, currentResult);
       } else {
         // compare current with merged and decide whether to merge the current result or create a new one
-        if (applySequentialGapBasedSplit
-            && (currentResult.getStartTime() - mergedAnomaly.getEndTime()) > sequentialAllowedGap) {
+        MergedAnomalyResultDTO currMergedAnomaly = new MergedAnomalyResultDTO();
+        populateMergedResult(currMergedAnomaly, currentResult);
+        if ((applySequentialGapBasedSplit
+            && (currentResult.getStartTime() - mergedAnomaly.getEndTime()) > sequentialAllowedGap)
+            || (!isMergeable(mergedAnomaly, currMergedAnomaly))) {
```

Comment:
Keep the logic as it is here, and change the function isMergeable to isEqualOnMergeableKeys.
This if-statement is doing a comparison on sequentialAllowedGap and if is equal on mergeable keys.
See line 109 is comparing on maxMergedDurationMillis
The AnomalyMergedConfig is fully used in this block of code, to really extract the isMergeable function needs to refine the whole block rather than simply this if-statement, hence changed the function name to specify the specific condition.